support for sles-12 and redhat7

pguibert6WIND opened the pull request:
hello @maheshloni ,
i would like you to review this pull request.


since i replace facter tool with lsb_release.

regarding the usage of facter, it is a mess to install it on sles-12.
that is why we want to propose using lsb_release instead.

Can I ask you to check if lsb_release is installed on other distribs ?
Thanks

xiaofeng6WIND committed:
pkgsrc: adapt dev_compile script for redhat distribution

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: reduce the call of facter command

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: adapt dev_compile script for suse distribution

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

maheshloni commented:
Hi Philippe,
lsb_release  command  doesn't work in Centos7. More over CSIT accepts only using facter command (standard command) 

Note: CSIT  suggested standard command to fetch OS and its version from various flavors of operating system instead of using lsb_release or uname -a 

I think its better to add one more option for sles12 : zypper install facter  by appending in the below line #60 :
         apt-get install -y facter || yum -y install facter || < sles12 command to install facter>   ;

 and for redhat os : yum command will work for installing the facter .
Can you please check on sles12 whether zypper install facter is works or not  . If you know proper syntax to install the facter command in sles12 , then its better to add it.

xiaofeng6WIND committed:
pkgsrc: package quagga and zrpc

RPM/DEB files will be generated and placed under pkgsrc directory.

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

pguibert6WIND commented:
Hi @maheshloni ,
a new rework of the script is available here.
please review.

pguibert6WIND commented:
Hi @maheshloni , please review the pull request.
some update has been done.

pguibert6WIND commented:
use following command to test:

./pkgsrc/dev_compile_script.sh -p -d -b -v 5

xiaofeng6WIND committed:
pkgsrc: rename package file

Add the following information into package name:
- the last commit id
- distribution and release version

Also, remove the date from package name.

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: add logrotate script

This patch adds zrpcd_log_rotate.sh script which is called by
init.d/zrpcd.

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: support systemd for RedHat distribution

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND commented:
Hi @maheshloni , please review new patches:
- last commited id and distribution is added into package name
- support log rotation
- support systemd for Redhat

Thanks.

pguibert6WIND committed:
pkgsrc: centos packaging was not creating username quagga

This fix is removing the -c parameter used when creating user quagga.
This permits Centos to install zrpc and quagga packages with the quagga
account.

Signed-off-by: Philippe Guibert <philippe.guibert@6wind.com>

maheshloni reviewed (approved):
will test this

xiaofeng6WIND committed:
pkgsrc: package c-capnproto

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: package thrift

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND committed:
pkgsrc: package zmq

Signed-off-by: Xiaofeng Liu <xiaofeng.liu@6wind.com>

xiaofeng6WIND commented:
Hi @pguibert6WIND @maheshloni  , Please review the patches:
1. pkgsrc: package c-capnproto
2. pkgsrc: package thrift
3. pkgsrc: package zmq

pguibert6WIND commented on pkgsrc/packaging.sh:
```diff
@@ -304,6 +304,93 @@ quagga_deb_bin_control () {
+
+ccapnproto_rpm_bin_spec () {
+    echo "Name: c-capnproto" >> $RPM_SPEC_FILE
+    if [ -z "$COMMITID" ]; then
```
I have one remark:
- c-capnproto is version 1.0.2. And it almost never changes.
I would say that the packging name should not change. => no commit ID.

pguibert6WIND commented on pkgsrc/packaging.sh:
```diff
@@ -391,6 +391,111 @@ ccapnproto_deb_bin_control () {
+
+thrift_rpm_bin_spec () {
+    echo "Name: thrift" >> $RPM_SPEC_FILE
+    if [ -z "$COMMITID" ]; then
```
I don' think it is needed to package thrift under commit ID naming specific too

pguibert6WIND reviewed (commented):
some comments on the packaging name for c-capnproto, zmq, and thrift.

maheshloni reviewed (commented):
will test and update
